cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -223,27 +249,37 @@ func (m *McmManager) GetMachineDeploymentForMachine(machine *Ref) (*MachineDeplo
 	}, nil
 }
 
-//Cleanup does nothing at the moment.
-//TODO: Enable cleanup method for graceful shutdown.
+// Refresh refreshes the lister caches
+func (m *McmManager) Refresh() error {
+	var stopCh <-chan struct{}
```

Comment:
`stopCh` is never closed. Ideally, this should be linked with the termination of the main routine (signal handler channel?).

Reply:
This is the function that invokes the cloud provider creation - https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/cloudprovider/builder/builder_all.go#L77. I don't see any channels being passed from here. So not sure what's the best way here. I have closed the channel by [defer for now](https://github.com/gardener/autoscaler/pull/73/commits/a7d47db2a052a8b144e361afe5dae97e9aeea612#diff-fc87dde41e1a7abff0b0bc24dbabc258ac3d594a346b8f01e020b9e7f9775440R173). Please recommend if we can do any better.

Reply:
Hmm... the signal handling in non-existent even in the main function and the [leader election loss handling](https://github.com/gardener/autoscaler/blob/7b07db383f0c644f16eb8e59c1614e9a22b9103a/cluster-autoscaler/main.go#L434) is also quite amusing :-)

I guess this is the best we can do for now.